crates/db_schema/src/traits.rs

```diff
@@ -158,29 +158,100 @@ pub trait Saveable {
   fn save(
     pool: &mut DbPool<'_>,
     form: &Self::Form,
-  ) -> impl Future<Output = Result<Self, Error>> + Send
+  ) -> impl Future<Output = LemmyResult<Self>> + Send
   where
     Self: Sized;
   fn unsave(
     pool: &mut DbPool<'_>,
     form: &Self::Form,
-  ) -> impl Future<Output = Result<uplete::Count, Error>> + Send
+  ) -> impl Future<Output = LemmyResult<uplete::Count>> + Send
+  where
+    Self: Sized;
+}
+
+pub trait Readable {
```

Comment:
Added a few more traits like Readable, Hideable, to be consistent with the other actions.